k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -1489,12 +1533,28 @@ public void onMessageBuildSuccess(MimeMessage message, boolean isDraft) {
             }
         } else {
             currentMessageBuilder = null;
+
+            if (mAction == Action.REDIRECT) {
+                LocalMessage referenceMessage = mMessageReference.restoreToLocalMessage(getApplicationContext());
+                message = referenceMessage.cloneAsSuper(true);
+                message.setResentDate(new Date(), K9.hideTimeZone());
+                message.setRecipients(RecipientType.RESENT_TO, toArray(recipientPresenter.getToAddresses()));
```

Comment:
Maybe I'm just missing something here. Just to help me understand, at the end of this block of code the message that is queued for sending will be the originally referenced one, except with added `Resent-` headers, right? But the addresses where the message will be actually delivered are the addresses in `To`, `Cc` and `Bcc` (see `SmtpTransport`), so doesn't that mean it will be sent to the same recipients as original one, rather than the ones they should be redirected to?